src/__tests__/TextInput.test.tsx

```diff
@@ -57,10 +57,56 @@ describe('TextInput', () => {
 
   it('renders leadingVisual', () => {
     expect(render(<TextInput name="search" placeholder={'Search'} leadingVisual={SearchIcon} />)).toMatchSnapshot()
+    expect(render(<TextInput name="search" placeholder={'Search'} leadingVisual={<SearchIcon />} />)).toMatchSnapshot()
```

Comment:
just a bunch of tests that validate rendered content given a bunch of different element types